addons/account/models/account_payment.py

```diff
@@ -853,6 +860,15 @@ def _check_payment_method_line_id(self):
             elif pay.payment_method_line_id.journal_id and pay.payment_method_line_id.journal_id != pay.journal_id:
                 raise ValidationError(_("The selected payment method is not available for this payment, please select the payment method again."))
 
+    @api.onchange('payment_method_line_id')
+    def _onchange_journal(self):
```

Comment:
I think normally onchanges are named after the field they're triggered by (here `payment_method_line_id`)